apps/postgres.go

```diff
@@ -0,0 +1,274 @@
+unlabeledNodes, err := SetupPostgresEnvironment()
+if err != nil {
+    log.Fatalf("Failed to set up environment: %v", err)
+}
+
+// Step 2: Define storage class name and volume name
+
+// Step 3: Build the PostgreSQL application configuration with PgBench
+postgresApp, err := apps.NewPostgresBuilder().
+    WithPgBench().
+    WithOwnStorageClass(scName).
+    WithPvc(volName).
+    WithPrimaryNodeSelector(unlabeledNodes[0].Name).
+    Build()
+if err != nil {
+    log.Fatalf("Failed to build PostgreSQL app: %v", err)
+}
+
+// Step 4: Initialize PgBench
+err = postgresApp.PgBench.InitializePgBench(fmt.Sprintf("%s-postgresql", postgresApp.Postgres.ReleaseName))
+if err != nil {
+    log.Fatalf("Failed to initialize PgBench: %v", err)
+}
+
+// Step 5: Run PgBench in a goroutine to perform load testing
+var wg sync.WaitGroup
+wg.Add(1)
+go func() {
+    defer wg.Done()
+    err := postgresApp.PgBench.RunPgBench(fmt.Sprintf("%s-postgresql", postgresApp.Postgres.ReleaseName))
+    if err != nil {
+        log.Fatalf("Failed to run PgBench: %v", err)
+    }
+}()
+
+// Step 6: Wait for PgBench job to complete
+wg.Wait()
+logf.Log.Info("pgBench job completed")
+*/
+
+type postgresBuilder struct {
+	architecture            Architecture
+	CloneFsIdAsVolumeIdType common.CloneFsIdAsVolumeIdType
+	filesystemType          common.FileSystemType
+	helmVersion             string
+	namespace               string
+	nodeSelector            string
+	pgBench                 bool
+	provisioningType        common.ProvisioningType
+	pvcName                 string
+	pvcSize                 int
+	releaseName             string
+	replicaCount            int
+	scName                  string
+	values                  map[string]interface{}
+}
+
+type PostgresApp struct {
+	Postgres k8stest.PostgresApp
+	PgBench  k8stest.PgBenchApp
+}
+
+type Limits struct {
+	Memory int `json:"memory"`
+	Cpu    int `json:"cpu"`
+}
+
+// NewPostgresBuilder creates new postgres builder with default standalone settings
+func NewPostgresBuilder() *postgresBuilder {
+	def := make(map[string]interface{})
+	def["global.postgresql.auth.postgresPassword"] = e2e_config.GetConfig().Product.PostgresAuthRootPassword
+	def["global.postgresql.auth.username"] = e2e_config.GetConfig().Product.PostgresAuthUsername
+	def["global.postgresql.auth.password"] = e2e_config.GetConfig().Product.PostgresAuthPassword
+	def["global.postgresql.auth.database"] = e2e_config.GetConfig().Product.PostgresDatabaseName
+	def["architecture"] = Standalone.String()
+	def["replicaCount"] = 1
+
+	return &postgresBuilder{
+		architecture:            Standalone,
+		CloneFsIdAsVolumeIdType: common.CloneFsIdAsVolumeIdNone,
+		filesystemType:          common.Ext4FsType,
+		namespace:               common.NSDefault,
+		provisioningType:        common.ThinProvisioning,
+		releaseName:             e2e_config.GetConfig().Product.PostgresReleaseName,
+		replicaCount:            0,
```

Comment:
What does it mean when replicaCount is set to 0 ?

Reply:
It is used as a tag for the automatic creator of the storage class if we want to automatically set replicas equal to the pool count if replicaCount == 0 or the user sets a custom value if it is > 0. However, if we also count on using it without mayastore, there is no need to maintain this logic and we could remove it.

Reply:
Removed